pkg/apis/core/validation/shoot_test.go

```diff
@@ -1889,54 +1890,71 @@ var _ = Describe("Shoot Validation Tests", func() {
 				Entry("invalid > 1 threshold", core.ClusterAutoscaler{
 					ScaleDownUtilizationThreshold: pointer.Float64(1.5),
-				}, ConsistOf(field.Invalid(field.NewPath("scaleDownUtilizationThreshold"), 1.5, "can not be greater than 1.0"))),
+				}, supportedVersionForIgnoreTaints, ConsistOf(field.Invalid(field.NewPath("scaleDownUtilizationThreshold"), 1.5, "can not be greater than 1.0"))),
 				Entry("valid with maxNodeProvisionTime", core.ClusterAutoscaler{
 					MaxNodeProvisionTime: &metav1.Duration{Duration: time.Minute},
-				}, BeEmpty()),
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
 				Entry("invalid with negative maxNodeProvisionTime", core.ClusterAutoscaler{
 					MaxNodeProvisionTime: &negativeDuration,
-				}, ConsistOf(field.Invalid(field.NewPath("maxNodeProvisionTime"), negativeDuration, "can not be negative"))),
+				}, supportedVersionForIgnoreTaints, ConsistOf(field.Invalid(field.NewPath("maxNodeProvisionTime"), negativeDuration, "can not be negative"))),
 				Entry("valid with maxGracefulTerminationSeconds", core.ClusterAutoscaler{
 					MaxGracefulTerminationSeconds: &positiveInteger,
-				}, BeEmpty()),
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
 				Entry("invalid with negative maxGracefulTerminationSeconds", core.ClusterAutoscaler{
 					MaxGracefulTerminationSeconds: &negativeInteger,
-				}, ConsistOf(field.Invalid(field.NewPath("maxGracefulTerminationSeconds"), negativeInteger, "can not be negative"))),
+				}, supportedVersionForIgnoreTaints, ConsistOf(field.Invalid(field.NewPath("maxGracefulTerminationSeconds"), negativeInteger, "can not be negative"))),
 				Entry("valid with expander least waste", core.ClusterAutoscaler{
 					Expander: &expanderLeastWaste,
-				}, BeEmpty()),
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
 				Entry("valid with expander most pods", core.ClusterAutoscaler{
 					Expander: &expanderMostPods,
-				}, BeEmpty()),
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
 				Entry("valid with expander priority", core.ClusterAutoscaler{
 					Expander: &expanderPriority,
-				}, BeEmpty()),
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
 				Entry("valid with expander random", core.ClusterAutoscaler{
 					Expander: &expanderRandom,
-				}, BeEmpty()),
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
+				Entry("valid with ignore taint", core.ClusterAutoscaler{
+					IgnoreTaints: ignoreTaintsUnique,
+				}, supportedVersionForIgnoreTaints, BeEmpty()),
+				Entry("duplicate ignore taint", core.ClusterAutoscaler{
+					IgnoreTaints: ignoreTaintsDuplicate,
+				}, supportedVersionForIgnoreTaints, ConsistOf(field.Duplicate(field.NewPath("ignoreTaints").Index(1), ignoreTaintsDuplicate[1]))),
+				Entry("invalid with ignore taint", core.ClusterAutoscaler{
+					IgnoreTaints: ignoreTaintsInvalid,
+				}, supportedVersionForIgnoreTaints, ConsistOf(metav1validation.ValidateLabelName(ignoreTaintsInvalid[0], field.NewPath("ignoreTaints").Index(0)))),
```

Comment:
Instead of running the validation itself (`ValidateLabelName`) just to be able to match the actual error, you could use the following:

```suggestion
				Entry("invalid with ignore taint",
					core.ClusterAutoscaler{
						IgnoreTaints: ignoreTaintsInvalid,
					},
					supportedVersionForIgnoreTaints,
					ConsistOf(PointTo(MatchFields(IgnoreExtras, Fields{
						"Type":  Equal(field.ErrorTypeInvalid),
						"Field": Equal("ignoreTaints[0]"),
					}))),
				),
```